Tests/MongoSwiftTests/MongoCursorTests.swift

```diff
@@ -0,0 +1,165 @@
+import Foundation
+@testable import MongoSwift
+import Nimble
+import NIO
+import TestsCommon
+
+private let doc1: Document = ["_id": 1, "x": 1]
+private let doc2: Document = ["_id": 2, "x": 2]
+private let doc3: Document = ["_id": 3, "x": 3]
+
+final class AsyncMongoCursorTests: MongoSwiftTestCase {
```

Comment:
These tests are basically the same as the sync versions but using the async driver API.